qutebrowser/browser/webkit/webkittab.py

```diff
@@ -50,6 +54,34 @@ def save_page(self):
         """Save the current page."""
         raise browsertab.UnsupportedOperationError
 
+    def show_source(self, win_id, url):
+
+        def format_url(url):
+            """emulate what WebEnginePage::ViewSource does."""
+            s = url.toString()
+            s = s.split('//')[-1]  # strip scheme
+            s = s.split('@')[-1]   # strip userinfo
+            return s
+
+        def show_source_cb(source):
+            """Show source as soon as it's ready."""
+            # WORKAROUND for https://github.com/PyCQA/pylint/issues/491
+            # pylint: disable=no-member
+            lexer = pygments.lexers.HtmlLexer()
+            formatter = pygments.formatters.HtmlFormatter(
+                full=True, linenos='table',
+                title='view-source:' + format_url(url))
+            # pylint: enable=no-member
+            highlighted = pygments.highlight(source, lexer, formatter)
+
+            base_url = QUrl('view-source:' + url.toString())
+            new_tab = tabbed_browser.tabopen(background=False, related=True)
+            new_tab.set_html(highlighted, base_url)
+
+        tabbed_browser = objreg.get('tabbed-browser', scope='window',
+                                    window=win_id)
+        tabbed_browser.currentWidget().dump_async(show_source_cb)
```

Comment:
If you store the tab in `AbstractAction`, this is just `self._tab.dump_async(show_source_cb)`.